src/main/java/net/wasdev/wlp/common/plugins/util/InstallFeatureUtil.java

```diff
@@ -0,0 +1,683 @@
+                return name.startsWith(INSTALL_MAP_PREFIX) && name.endsWith(INSTALL_MAP_SUFFIX);
+            }
+        });
+
+        File result = null;
+        if (installMapJars != null) {
+            for (File jar : installMapJars) {
+                if (isReplacementJar(result, jar)) {
+                    result = jar;
+                }
+            }
+        }
+
+        return result;
+    }
+    
+    /**
+     * Returns whether file2 can replace file1 as the install map jar.
+     */
+    private static boolean isReplacementJar(File file1, File file2) {
+        if (file1 == null) {
+            return true;
+        } else if (file2 == null) {
+            return false;
+        } else {
+            String version1 = extractVersion(file1.getName());
+            String version2 = extractVersion(file2.getName());
+            return compare(version1, version2) < 0;
+        }
+    }
+    
+    private static String extractVersion(String fileName) {
+        int startIndex = INSTALL_MAP_PREFIX.length() + 1;
+        int endIndex = fileName.lastIndexOf(INSTALL_MAP_SUFFIX);
+        if (startIndex < endIndex) {
+            return fileName.substring(startIndex, endIndex);
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Performs pairwise comparison of version strings, including nulls and non-integer components.
+     * @param version1
+     * @param version2
+     * @return positive if version2 is greater, negative if version1 is greater, otherwise 0
+     */
+    private static int compare(String version1, String version2) {
+        if (version1 == null && version2 == null) {
+            return 0;
+        } else if (version1 == null && version2 != null) {
+            return -1;
+        } else if (version1 != null && version2 == null) {
+            return 1;
+        }
+        String[] components1 = version1.split("\\.");
+        String[] components2 = version2.split("\\.");
+        for (int i = 0; i < components1.length && i < components2.length; i++) {
+            int comparison;
+            try {
+                comparison = new Integer(components1[i]).compareTo(new Integer(components2[i]));
+            } catch (NumberFormatException e) {
+                comparison = components1[i].compareTo(components2[i]);
+            }
+            if (comparison != 0) {
+                return comparison;
+            }
+        }
+        return components1.length - components2.length;
+    }
+    
+    /**
+     * Performs product validation by running bin/productInfo validate
+     * 
+     * @throws PluginExecutionException
+     *             if product validation failed or could not be run
+     */
+    public void productInfoValidate() throws PluginExecutionException {
+        Process pr = null;
+        InputStream is = null;
+        Scanner s = null;
+        Worker worker = null;
+        try {
+            String osName = System.getProperty("os.name", "unknown").toLowerCase();
```

Comment:
Can `isWindows` be moved to a general util class and made public?

Reply:
Moved to OSUtil class.